hibernate-core/src/main/java/org/hibernate/boot/model/source/internal/hbm/FilterDefinitionBinder.java

```diff
@@ -38,6 +39,7 @@ static void processFilterDefinition(
 			HbmLocalMetadataBuildingContext context,
 			JaxbHbmFilterDefinitionType jaxbFilterDefinitionMapping) {
 		Map<String, JdbcMapping> parameterMap = null;
+		Map<String, Class<? extends Supplier>> parameterResolverMap = null;
```

Comment:
I think we should resolve the managed beans already at this point i.e.

```suggestion
		Map<String, ManagedBean<Supplier>> parameterResolverMap = null;
```